File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int mDepth = 0;
    void func(TreeNode root, int depth){
        if(root == null) return;
        mDepth = Math.Max(mDepth, depth);
        func(root.left, depth+1);
        func(root.right, depth+1);
    }
    public int MaxDepth(TreeNode root) {
        func(root, 1);
        return mDepth;
    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int mDepth = 0;
    void func(TreeNode* root, int depth){
        if(root == NULL){
            return;
        }
        mDepth = max(mDepth, depth);
        func(root->left, depth+1);
        func(root->right, depth+1);
    }
    int maxDepth(TreeNode* root) {
        func(root, 1);
        return mDepth;
    }
};